[CS2113-F10-2] ClubInvMgr

quitejasper opened the pull request:
ClubInvMgr is a desktop CLI app for inventory management for CCA clubs, especially for fast typists who can accomplish tasks quickly by typing out commands.

heekit73098 commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,167 @@
+The `UI` component
+- Displays salutations, prompts for user input, error messages and results of queries.
+- Reads in user inputs
+- Depends on the `Messages` and `InvMgrException` classes in the `Common` component. It displays messages stored in the `Messages` class and displays an error message whenever `InvMgrException` is invoked.
```
Introducing this component out of the blue can be quite confusing, especially when InvMgrException is in the exceptions package instead of common

heekit73098 commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,167 @@
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
```
Consider adding an overall class diagram and introduce the major components so that it would not very confusing below when components are named suddenly.

tjiarong commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,167 @@
+## Implementation
+
+### Description Command
+![DescCommandSequenceDiagram](img/DescCommandSequenceDiagram.png)
+The following diagram shows the sequence diagram for retrieving the description of an item.
+
+For a user who is unaware of what an item is about, he/she can enter the command eg. `desc 2` command to extract the description for the second item in the inventory list. This command is interpreted by the `Parser` and a `DescCommand` is returned to `InvMgr`. `InvMgr` calls the execute command of `DescCommand` which retrieves the item's information from the `ItemList` and then outputs them into  ...
```
![image](https://user-images.githubusercontent.com/69455499/160962868-8797d01f-9d47-48e9-9263-737eaf8f1dfc.png)
In the UML diagram, the program calls writeData(ArrayList) at the end after the execution of the desc command. Not sure if this was intended as this was not explained the running of the desc command.

heekit73098 commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,167 @@
+a `Command` class based on the user input.
+
+### Command Component
+![CommandClassDiagram](img/CommandClassDiagram.png)
```
Consider enlarging this diagram as the words are significantly smaller than the other diagrams

cczhouqi replied:
You may also consider changing the way you arrange the boxes (to not in one line) to make the words larger but keep the diagram in a reasonable size.

tjiarong commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,167 @@
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ...                               | I want to ...                                   | So that I can ...                                            |
+|---------|----------------------------------------|-------------------------------------------------|--------------------------------------------------------------|
+| v1.0    | Manager                                | Add a new item to the inventory                 | Update my inventory                                          |
+| v1.0    | Manager                                | Remove an item from the inventory               | Update my inventory                                          |
+| v1.0    | Stocktaker                             | list out all my items                           | View all my items at a glance                                |
+| v1.0    | New user                               | List out all possible commands                  | I can familiarise myself with using the system               |
+| v1.0    | User who has not seen items physically | Get the description of a particular item        | I can visualise the item better to know if it is what i need |
+| v1.0    | As a frequent/first time user          | Write to a file containing the entire inventory | Save my inventory data to a file                             |
+| v1.0    | Stocktaker                             | Read from and load an inventory file data       | To work on and view the data                                 |
```
Perhaps you can update the user stories with those you have selected for V2.0?

heekit73098 commented on docs/diagrams/UiClassDiagram.puml:
```diff
@@ -0,0 +1,35 @@
+    + getRawUserInput(): String
+}
+
+InvMgr --> "1" Ui
```
Should this be a composition instead?

heekit73098 reviewed (commented):
Generally the DG is easy to read and understand

tjiarong commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,167 @@
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Application Launch
 
```
Consider adding a content page or navigation bar with hyperlink to improve navigability and document formatting

tjiarong commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,167 @@
+5. `DeleteCommand` will converse with `Ui` to show a message that the item has been removed. In this case, the item to add will be printed as the name of the item, followed by " has been deleted.".
+
+### List Command
+![ListCommandSequenceDiagram](img/ListCommandSequenceDiagram.png)
+
+The following diagram shows the sequence diagram of the listing of items in `itemList`.
+
+The user starts by typing a list command.
+
+1. `InvMgr` calls `parse("list")` method in `Parser` class, which returns a ListCommand object.
+2. `InvMgr` calls `execute(itemList, ui)` method in `ListCommand` object.
+3. `ListCommand` loops through every `Item` in `itemList` and prints them line by line
+   and numbers them.
```
![image](https://user-images.githubusercontent.com/69455499/160964986-c78cea65-4825-450e-b147-f44cea5b9f24.png)
Consider moving the end of lifeline cross out of the object box and at the end of the line.

tjiarong commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,167 @@
-{Describe the value proposition: what problem does it solve?}
+1. Centralised management of resources that ensures accurate and timely allocation of equipment to students
+2. Increases the ease and efficiency of resource management
+3. More organised
```
Consider elaborating on this value proposition as it is quite vague. Could possible extend to be similiar like point 1.

yanjie1017 commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,167 @@
+
+### List Command
+![ListCommandSequenceDiagram](img/ListCommandSequenceDiagram.png)
+
```
![Screenshot 2022-03-31 at 11 22 16](https://user-images.githubusercontent.com/60136330/160969977-5948d42f-6a66-46f8-990c-1be68bfc17bb.png)
Should the lifeline of the object carry on if the class has been terminated?

yanjie1017 commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,167 @@
+### Parser Component
+
+![ParserClassDiagram](img/ParserClassDiagram.png)
+
```
![Screenshot 2022-03-31 at 11 27 54](https://user-images.githubusercontent.com/60136330/160970511-5ef013f4-87ba-4791-be7f-ff7be16fe6ee.png)
Consider stating the return types of the methods called in InvMgr?

yanjie1017 replied:
The format of attributes should be 
NameOfAttribute : DataTypeOfAttribute

cczhouqi commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,167 @@
+6. `AddCommand` will converse with `Ui` to show a message that the item has been added. In this case, the item to add will be printed as the name of the item, followed by " has been added!".
+
+### Delete Command
+![DeleteCommandSequenceDiagram](img/DeleteCommandSequenceDiagram.png)
```
Some of the diagrams are above the "The following diagram" line while others are below. You may consider keeping it consistent :D

cczhouqi reviewed (commented):
There are some minor things that can be improved (as shown in others' comments also), but generally it looks quite good 👍🏻

yanjie1017 commented on docs/DeveloperGuide.md:
```diff
@@ -4,26 +4,167 @@
+
+### Description Command
+![DescCommandSequenceDiagram](img/DescCommandSequenceDiagram.png)
+The following diagram shows the sequence diagram for retrieving the description of an item.
```
![Screenshot 2022-03-31 at 11 34 48](https://user-images.githubusercontent.com/60136330/160971269-26fd4a13-79fe-4947-a8e0-011a155290d2.png)
![Screenshot 2022-03-31 at 11 34 58](https://user-images.githubusercontent.com/60136330/160971302-ecfdda2e-861c-49ee-bc81-02bdb5691bda.png)
Inconsistent format. Can decide whether to add the footer.

yanjie1017 reviewed (commented):
Looks good in overall. May require some minor changes and add in the remaining sections.

thernturnaround committed:
Change output of SearchCommand to use original index of returned Item

thernturnaround committed:
Fix Item print bug

lestersimjj committed:
Update BorrowCommand to not allow borrowing of lost items

thernturnaround committed:
Change Storage to use an adapter for BorrowRecord instead of just LocalDate

BorrowStatus had no way of being updated before.

To keep the listob, listfb commands relevant to the current date,

Let's
- Write an adapter for BorrowRecord
- Update the BorrowStatus everytime Storage loads in data

lestersimjj committed:
Update Item class to track total quantity borrowed instead of overlapping dates.

thernturnaround committed:
Add BorrowRecords to Storage tests
Refactor CommandStubs to ItemList Stubs
Added equals() check for BorrowRecords, needed for equality check in Storage tests

lestersimjj committed:
Update to UserGuide for borrow quantity

lestersimjj committed:
Update ParserUtils to specify DateTimeFormatter.

This is to help resolve the JSON serializer issue when storing the data.

lestersimjj committed:
Update BorrowCommandParser to enforce max 7 days borrowing.

lestersimjj committed:
Replace magic number for max borrow duration with constant

thernturnaround committed:
Update generateItemList for use in stubs

Java, by default, adds references of objects into arraylist. This means the original Items in ItemList may be modified.

This may result in a race condition! Each test are given their own threads to execute in JUnit.

To fix this, we should clone the Item and add them to the list before returning the itemlist.

Let's
- Implement clone() for BorrowRecord, needed for deep copying
- Implement a deep copy of Item() in its clone method to clone BorrowRecord
- Change ItemListStubs to generate new mutable ItemLists that are clones of the specified Items.

thernturnaround committed:
Fix checkstyle errors

thernturnaround committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into branch-general-bugfix

thernturnaround committed:
Update gitignore

thernturnaround committed:
Fix a missed merge conflict marker

lestersimjj committed:
Merge pull request #181 from lestersimjj/branch-Borrow

Update BorrowRecord and BorrowCommand

thernturnaround committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into branch-general-bugfix
Adapted new code into BorrowRecordAdapter

thernturnaround committed:
Fix test code to work with changes to borrowrecord

thernturnaround committed:
Remove actual output test files

thernturnaround committed:
Remove clone() usage.

Cloneable and clone is frowned upon.

Let's remove its usages and create manual copies through a copyItem()/copyBorrowRecord() instead.

thernturnaround committed:
Catch NullPointerException when parsing goes wrong

thernturnaround committed:
Remove clone test

thernturnaround committed:
Migrate ItemList Stubs to the CommandTest they are used in

Some weird, undiagnosable behaviour happened when the lists were globally defined in ItemListStubs.

Defining them locally (within the CommandTest) seems to resolve the issue

thernturnaround committed:
Update expected data for storage test

thernturnaround committed:
Fix checkstyle errors

Commit:
Adds ReturnCommandTest

Commit:
Merge branch 'branch-ReturnCommand' of https://github.com/chihyingho/tp into branch-ReturnCommand

chihyingho committed:
Merge pull request #174 from chihyingho/ChihYing-PPP

Adds first draft of chihyingho's PPP

thernturnaround committed:
Merge pull request #167 from IncompetentDev/branch-general-bugfix

PE-D bugfixes

quitejasper committed:
Update ListFutureBorrowingsCommand.java

Resolves #160

Commit:
Fixes #126

Adds ReturnCommandParserTest and ReturnCommandTest

chihyingho committed:
Merge branch 'master' into branch-ReturnCommand

Commit:
Fixes checkstyle rule violations

Commit:
Merge branch 'branch-ReturnCommand' of https://github.com/chihyingho/tp into branch-ReturnCommand

Commit:
Modifies ReturnCommandTest to handle modified BorrowRecord constructor

Commit:
Fixes checkstyle rule violations

Commit:
Fixes checkstyle rule violations

Commit:
Fixes import error

Commit:
Merge branch 'branch-LostCommand' of https://github.com/chihyingho/tp into branch-LostCommand

Commit:
Fixes #178

Modifies LostCommand to either reduce the quantity of a lost item or delete it entirely

lestersimjj committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into branch-ListCurrentBorrowingsCommand

quitejasper committed:
Update ListFutureBorrowingsCommand.java

Resolves #151

Programme was initially unable to detect when there are no relevant
records in the future (and for the particular borrower). This does not
align with the User Guide where we should be able to.

Let's
* Use Java Streams to create a compiled list of relevant borrow records,
and use that list to check if we should return a message saying that
there are no records.

quitejasper committed:
Update ListCommand.java

Tells the user that there are no items when it checks there there are
no items in the item list.

Commit:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into branch-LostCommand

Commit:
Modified BorrowCommand to remove references to deleted isLost attribute of Item

chihyingho committed:
Merge pull request #186 from chihyingho/branch-ReturnCommand

Fixes #126

lestersimjj committed:
Update Item filterRecords method to map item name when printing

Commit:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into branch-LostCommand

thernturnaround committed:
Add UML Sequence diagrams to reflect changes
Changes/additions: EditCommand, SearchCommand, Storage Load/Save

thernturnaround committed:
Update DG with architecture details

thernturnaround committed:
Edit DeveloperGuide.md

thernturnaround committed:
Change order of ApplicationStartup seq diagram

quitejasper committed:
Merge pull request #189 from lestersimjj/branch-ListCurrentBorrowingsCommand

Update Item in filtering records

quitejasper committed:
Merge branch 'AY2122S2-CS2113-F10-2:master' into master

quitejasper committed:
Merge branch 'master' into branch-ListFutureBorrowingsCommand

thernturnaround committed:
Add Parser Component to DG

Commit:
Adds LostCommandTest

Commit:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into branch-LostCommand

quitejasper committed:
Update ListFutureBorrowingsCommand.java

`Item` already has an inbuilt function to retrieve a list of strings
that represent the BorrowRecords.

Let's:
* Leverage on this function in retrieving our borrowrecords (with the
name of the item)
* Use flatmap to collate all borrowrecords so as to determine whether
there are any borrowings at all
* Modify remaining code to utilise the list of records created, rather
than creating another list for the sake of printing

quitejasper committed:
Update ListCurrentBorrowingsCommand.java

Modified execute method in accordance to ListFutureBorrowingsCommand.

Context:
`Item` already has an inbuilt function to retrieve a list of strings
that represent the BorrowRecords.

Let's:
* Leverage on this function in retrieving our borrowrecords (with the
name of the item)
* Use flatmap to collate all borrowrecords so as to determine whether
there are any borrowings at all
* Modify remaining code to utilise the list of records created, rather
than creating another list for the sake of printing

quitejasper committed:
Add expectedData.json

thernturnaround committed:
Add Storage Component to DG

quitejasper committed:
Test formatting of code in ListFutureBorrowingsCommand.java

Testing for checkstyle errors if we align methods.

quitejasper committed:
Update ListCurrentBookingsCommand.java

Reformmatted code to align

quitejasper committed:
Merge pull request #185 from quitejasper/branch-ListFutureBorrowingsCommand

Update ListFutureBorrowingsCommand.java

quitejasper committed:
Merge pull request #188 from quitejasper/master

Update ListCommand.java and ListCurrentBorrowingsCommand.java

thernturnaround committed:
Update NFR in DG

quitejasper committed:
Merge pull request #191 from quitejasper/A-Data

Add expectedData.json

quitejasper committed:
Update InvMgr.java

Resolves #128

Added divider for cosmetic changes

cheshire-doge committed:
Fix listab startDate after endDate

cheshire-doge committed:
Update list in UG to let users know arguments after list are disregarded

quitejasper committed:
Update InvMgr.java and Expected.txt

Removed assertion statement at the end of run method. Added divider in
test file.

thernturnaround committed:
Update EditCommand in DG

quitejasper committed:
Merge branch 'AY2122S2-CS2113-F10-2:master' into master

thernturnaround committed:
Add Search Command implementation to DG

thernturnaround committed:
Update Storage implementation in DG

thernturnaround committed:
Fix error in referring to save() method of Storage in DG

thernturnaround committed:
Normalise UserGuide format

quitejasper committed:
Update expectedData.json

Reformatted json file for easy viewing

quitejasper committed:
Update ListOverdueBorrowingsCommand.java

Compared to future and current borrowings, overdue borrowings require an
additional check for items where items have not been returned.

* New method filterOverdueRecords in `Item` class for the additional
filter for items where isReturned == false
* execute method in command class is modified accordingly

quitejasper committed:
Update code to match checkstyle

quitejasper committed:
Update JavaDoc

thernturnaround committed:
Clean up & document code
- Removed unused imports
- Tidy up existing documentation to fit convention
- Migrated some constant strings to common/Messages
- Update HELP_MESSAGE for SearchCommand and EditCommand

Fixes #161

thernturnaround committed:
Fix checkstyle errors

thernturnaround committed:
Clean up and Document test code

thernturnaround committed:
Fix checkstyle errors

cheshire-doge committed:
Fix listab to show quantity available for borrowings

quitejasper committed:
Merge pull request #192 from quitejasper/master

Update InvMgr.java

cheshire-doge committed:
Modify magic strings and

quitejasper committed:
Revert "Update expectedData.json"

This reverts commit 22f04ce529659021459245b4f1490432f5776e15.

thernturnaround committed:
Fix static variable dataPath in Storage

quitejasper committed:
Update List Borrowings Commands

Added numbering for CancelFutureBooking

quitejasper committed:
Update ListOverdueBookingsCommand.java

Added numbering

quitejasper committed:
Merge pull request #197 from quitejasper/branch-ListFutureBorrowingsCommand

Update List Borrowings Commands

quitejasper committed:
Merge pull request #195 from quitejasper/A-Data

Minor changes, no update to expectedData.json

quitejasper committed:
Merge pull request #196 from quitejasper/branch-ListOverdueBorrowingsCommand

Update ListOverdueBookingsCommand.java

thernturnaround committed:
Revert static var List<Item> in ItemList
Change LostCommand to utilise non-static context in accessing ItemList

thernturnaround committed:
Refactor AddCommandTest to use provided ItemStubs

quitejasper committed:
Update AddCommand.java

Add Command allowed for duplicate addition of items which should not be
the case.

Let's:
* Add a check to see if there is an item with the same name in the list
* Direct users to use Edit Command if there is a matched item in the
list

quitejasper committed:
Remove Redundant Imports from AddCommand.java

quitejasper committed:
Update AddCommandParser.java

Resolves #194

Zero quantity could be added into the inventory.

Let's:
* Catch this error during the parsing stage of AddCommand
* Return exception if the quantity is found to be zero or negative

quitejasper committed:
Update AddCommandParser.java JavaDoc

quitejasper committed:
Update AddCommandParser.java

`ParserUtils.java` already has a check for negative quantity.

Let's:
* Remove check for negative quantity in the later stages of creating
AddCommand object

quitejasper committed:
Fix checkstyle errors

cheshire-doge committed:
Add exceptions and assertions to CancelFutureBorrowings, additional fixes to listab

Commit:
Adds LostCommandParserTest

Commit:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into branch-DeveloperGuide

cheshire-doge committed:
Merge branch 'A-Final'

Commit:
Fixes checkstyle rule violations

cheshire-doge committed:
Minor fix to javadocs

cheshire-doge committed:
Fixes to checkstyle

chihyingho committed:
Merge pull request #187 from chihyingho/branch-LostCommand

Modifies LostCommand to edit a lost item's quantity or delete it altogether

Commit:
Adds an additional InvMgrException

cheshire-doge committed:
Merge pull request #199 from cheshire-doge/master

Fix ListAvailableBorrowings, modifications to CancelFutureBorrowings

thernturnaround committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into branch-cleanup-and-documentation

thernturnaround committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into branch-ppp

thernturnaround committed:
Merge pull request #190 from IncompetentDev/branch-cleanup-and-documentation

Updates for DG, UG and codebase

thernturnaround committed:
Update ppp

thernturnaround committed:
Update ppp

thernturnaround committed:
Update ppp

quitejasper committed:
Update DeleteCommand

* Modified class code
* Added testing for DeleteCommand

lestersimjj committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into branch-ListCurrentBorrowingsCommand

quitejasper committed:
Merge branch 'master' into branch-AddCommand

# Conflicts:
#	src/main/java/seedu/duke/common/Messages.java

lestersimjj committed:
Update HelpCommand to include help messages from other commands

From EditCommand, ListCurrentBorrowingsCommand, ListFutureBorrowingsCommand, ListOverdueBorrowingsCommand

quitejasper committed:
Update AddCommandParser and AddCommand Tests

`AddCommandTest`
* Modified code
* Included test for duplicate items

`AddCommandParserTest`
* Added tests for zero and negative quantity

quitejasper committed:
Add JavaDoc to AddCommandTest.java

cheshire-doge committed:
Update Developer Guide for ListCommand, ListAvailableBorrowingsCommand and CancelFutureBorrowings Command

thernturnaround committed:
Update UG

quitejasper committed:
Merge pull request #201 from quitejasper/branch-DeleteCommand

Update DeleteCommand

Commit:
Updates Developer Guide with ReturnCommand implementation description

lestersimjj committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into A-DeveloperGuide

Commit:
Refines LostCommand

lestersimjj committed:
Merge pull request #202 from lestersimjj/branch-ListCurrentBorrowingsCommand

Update HelpCommand with messages from other Commands

quitejasper committed:
Refactor Duplicate Item Message

Moved message from execute method to common/Messages

quitejasper committed:
Update Messgaes

added newline

lestersimjj committed:
Update DescCommandSequenceDiagram to remove footbox for standardization

Commit:
Updates DG with implementation details of LostCommand

Commit:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp

lestersimjj committed:
Update DescCommand sequence diagram and DG

quitejasper committed:
Update UserGuide.md

quitejasper committed:
Update UserGuide.md

cheshire-doge committed:
Modify magic strings, update HelpCommand

cheshire-doge committed:
Merge branch 'A-DeveloperGuide'

Commit:
Updates UG with information on ReturnCommand, LostCommand and HelpCommand

lestersimjj committed:
Update CommandClassDiagram and DG

Commit:
Updates the UG list of features

Commit:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into ChihYing-PPP

quitejasper committed:
Merge pull request #198 from quitejasper/branch-AddCommand

Update AddCommand.java

thernturnaround committed:
Change EditCommand functionality

There is an issue with the logic related to booking items if EditCommand is able to decrease the quantity of an item.

EditCommand has been changed to only be able to increase the quantity of items.

Let's
- Remove the relative prefix
- Remove parseRelative() from parserUtils
- Change the code accordingly in EditCommandParser to not look for the relative prefix
- Change the code accordingly in EditCommand to default to relative addition
- Update test for EditCommand and EditCommandParser

thernturnaround committed:
Fix checkstyle errors

thernturnaround committed:
Update UG and DG to reflect the changes

cheshire-doge committed:
Modifications to DG

cheshire-doge committed:
Merge branch 'AY2122S2-CS2113-F10-2:master' into master

thernturnaround committed:
Fix hyperlink error for images

lestersimjj committed:
Add CommandClassDiagram2 and CommandClassDiagram3

Commit:
Updates chihyingho'2 PPP

Commit:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into branch-LostCommand

Commit:
Modifies LostCommand such that item is wholly deleted from inventory when marked lost

LostCommandParser and relevant JUnit tests were modified as well

Commit:
Fixes #203

quitejasper committed:
Update DeveloperGuide.md for Jasper

Commit:
Update DG LostCommand section to reflect a modified LostCommand that deletes a lost item from the inventory

thernturnaround committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into branch-cleanup-and-documentation

Commit:
Fixes #133

Adds punctuation behind each parameter description in JavaDoc

chihyingho committed:
Merge pull request #212 from chihyingho/ChihYing-PPP

Updates PPP

Commit:
Refines ReturnCommand

Commit:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into branch-HelpCommand

quitejasper committed:
Update DG

quitejasper committed:
Update DG images

Commit:
Modifies help message in all command classes and ensures all are included in HelpCommand

Commit:
Fixes #173

chihyingho committed:
Merge pull request #205 from chihyingho/branch-LostCommand

Refines LostCommand

Commit:
Fixes checkstyle rule violations

Commit:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into branch-DeveloperGuide

lestersimjj committed:
Add ItemListClassDiagram

lestersimjj committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into A-UserGuide

cheshire-doge committed:
Update PPP

chihyingho committed:
Merge pull request #214 from chihyingho/branch-Ui

Fixes #133

chihyingho committed:
Merge pull request #215 from chihyingho/branch-ReturnCommand

Refines ReturnCommand

lestersimjj committed:
Update to UG

Commit:
Updates DG with information on error handling for ReturnCommand and LostCommand

Commit:
Modifies ReturnCommand to throw additional exception

cheshire-doge committed:
runtest fix

lestersimjj committed:
Updates to DG to remove unnecessary lines

lestersimjj committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into A-PPP

lestersimjj committed:
Updates to PPP

cheshire-doge committed:
change expected.txt

lestersimjj committed:
Updates to PPP lestersimjj

cheshire-doge committed:
more change to expected.txt

cheshire-doge committed:
Change messages

quitejasper committed:
Update DG with ListFutureBorrowings

lestersimjj committed:
Updates to PPP

quitejasper committed:
Update DG with ListOverdueBorrowings

thernturnaround committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into branch-ppp

lestersimjj committed:
Updates to UG examples

cheshire-doge committed:
Change expected.txt

Commit:
Updates DG with information on HelpCommand implementation and also modifies LostCommand and ReturnCommand implementation descriptions

cheshire-doge committed:
fix runtest

Commit:
Fixes typos and checkstyle rule violations

Commit:
Updates PPP

lestersimjj committed:
Add error handling into DG for DescCommand

thernturnaround committed:
Update PPP

cheshire-doge committed:
Update UG

thernturnaround committed:
Add warning for editing data file

cheshire-doge committed:
Further modification to UG

Commit:
Updates DG with new sequence diagrams and description for LostCommand and ReturnCommand and HelpCommand

cheshire-doge committed:
Merge pull request #210 from cheshire-doge/master

Update DG with CancelFutureBorrowings and ListAvailableBorrowings

chihyingho committed:
Merge pull request #218 from chihyingho/branch-ReturnCommand

Modifies ReturnCommand to throw additional exception

chihyingho committed:
Merge pull request #208 from chihyingho/branch-UserGuide

Updates the UG with information on ReturnCommand, LostCommand and HelpCommand

chihyingho committed:
Merge branch 'master' into branch-HelpCommand

lestersimjj committed:
Add BorrowCommandSequenceDiagram

quitejasper committed:
Update Jasper PPP (Final)

thernturnaround committed:
Merge pull request #209 from IncompetentDev/branch-cleanup-and-documentation

Fix for #203

quitejasper committed:
Add Hyperlinks within PPP

quitejasper committed:
Minor Fixes

Commit:
Updates PPP

lestersimjj committed:
Update to DG on BorrowCommandSequenceDiagram

chihyingho committed:
Merge pull request #216 from chihyingho/branch-HelpCommand

Fixes #173

quitejasper committed:
Add ListFutureBookingsTest

quitejasper committed:
Update PPP

quitejasper committed:
Merge pull request #207 from quitejasper/A-UserGuide

Update UserGuide.md

chihyingho committed:
Merge pull request #204 from chihyingho/branch-DeveloperGuide

Updates DG with ReturnCommand, LostCommand and HelpCommand implementation details

lestersimjj committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into A-UserGuide

# Conflicts:
#	docs/UserGuide.md

lestersimjj committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into A-DeveloperGuide

# Conflicts:
#	docs/DeveloperGuide.md

lestersimjj committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into A-PPP

quitejasper committed:
Checkstyle fixes

Commit:
Merge branch 'master' of https://github.com/AY2122S2-CS2113-F10-2/tp into ChihYing-AboutUs

quitejasper committed:
checkstyle fixes

Commit:
Updates team's Portfolio links

cheshire-doge committed:
Update user stories

quitejasper committed:
Merge pull request #222 from quitejasper/branch-ListFutureBorrowingsCommand

Add ListFutureBookingsTest

lestersimjj committed:
Merge pull request #206 from lestersimjj/A-DeveloperGuide

Update DeveloperGuide

chihyingho committed:
Merge pull request #223 from chihyingho/ChihYing-AboutUs

Updates portfolio links

thernturnaround committed:
Add 2.1 to UG

lestersimjj committed:
Merge pull request #217 from lestersimjj/A-UserGuide

Update to UserGuide

cheshire-doge committed:
update PPP

Commit:
Modifies portfolio links

thernturnaround committed:
Trim lifeline

quitejasper committed:
Merge branch 'master' into A-DeveloperGuide

quitejasper committed:
Merge pull request #221 from quitejasper/A-PPP

Update Jasper PPP (Final)

chihyingho committed:
Merge pull request #225 from chihyingho/ChihYing-AboutUs

Modifies portfolio links

chihyingho committed:
Merge pull request #220 from chihyingho/ChihYing-PPP

Updates PPP

lestersimjj committed:
Update to lestersimjj PPP

cheshire-doge committed:
Merge pull request #224 from cheshire-doge/master

Update user stories for v2.1

lestersimjj committed:
Merge pull request #219 from lestersimjj/A-PPP

Updates to PPP: lestersimjj

thernturnaround committed:
Add TOC for DG

thernturnaround committed:
PPP update

quitejasper committed:
Merge pull request #213 from quitejasper/A-DeveloperGuide

Update DeveloperGuide.md for Jasper

thernturnaround committed:
Merge pull request #226 from IncompetentDev/branch-finaldgug

DG

thernturnaround committed:
Merge pull request #180 from IncompetentDev/branch-ppp

Final PPP

thernturnaround committed:
Update AboutUs.md

Update PPP link in AboutUs.md for IncompetentDev
